prow/tide/gerrit.go

```diff
@@ -89,15 +90,11 @@ func NewGerritController(
 		newPoolPending:   make(chan bool),
 	}
 
-	gitClient, err := (&flagutil.GitHubOptions{}).GitClientFactory(cookieFilePath, &configOptions.InRepoConfigCacheDirBase, false)
-	if err != nil {
-		logrus.WithError(err).Fatal("Error creating git client.")
-	}
-	cacheGetter, err := config.NewInRepoConfigCacheHandler(configOptions.InRepoConfigCacheSize, cfgAgent, gitClient, configOptions.InRepoConfigCacheCopies)
+	cacheGetter, err := config.NewInRepoConfigCacheHandler(configOptions.InRepoConfigCacheSize, cfgAgent, gc, configOptions.InRepoConfigCacheCopies)
 	if err != nil {
 		return nil, fmt.Errorf("failed creating inrepoconfig cache getter: %v", err)
 	}
-	provider := newGerritProvider(logger, cfgAgent.Config, nil, cacheGetter, cookieFilePath, "")
+	provider := newGerritProvider(logger, cfgAgent.Config, mgr.GetClient(), cacheGetter, cookieFilePath, "")
```

Comment:
I guess this mgr.GetClient() is the actual prowjob list client? The other stuff in this change is for the prow config caching, which I think is not related?

Reply:
yep

Reply:
the other change in this PR was btw cleanup

Reply:
Ack. It would be better to put those things in a separate commit for better `git blame` tracing in the future, please 🙏

Reply:
ack